core/src/main/scala/sttp/openai/json/SnakePickle.scala

```diff
@@ -45,20 +47,52 @@ object SttpUpickleApiExtension extends SttpUpickleApi {
   implicit def upickleBodySerializerSnake[B](implicit encoder: SnakePickle.Writer[B]): BodySerializer[B] =
     b => StringBody(SnakePickle.write(b), "utf-8", MediaType.ApplicationJson)
 
-  def asJsonSnake[B: SnakePickle.Reader: IsOption]: ResponseAs[Either[ResponseException[String, DeserializationException], B]] =
-    asString.mapWithMetadata(ResponseAs.deserializeRightWithError(deserializeJsonSnake)).showAsJson
+  def asJsonSnake[B: SnakePickle.Reader: IsOption]: ResponseAs[Either[OpenAIException, B]] =
+    asString.mapWithMetadata(deserializeRightWithMappedExceptions(deserializeJsonSnake)).showAsJson
 
-  def deserializeJsonSnake[B: SnakePickle.Reader: IsOption]: String => Either[DeserializationException, B] = { (s: String) =>
+  private def deserializeRightWithMappedExceptions[T](
+      doDeserialize: String => Either[DeserializationOpenAIException, T]
+  ): (Either[String, String], ResponseMetadata) => Either[OpenAIException, T] = {
+    case (Left(body), meta) =>
+      Left(httpToOpenAIError(HttpError(body, meta.code)))
+    case (Right(body), _) => doDeserialize.apply(body)
+  }
+
+  def deserializeJsonSnake[B: SnakePickle.Reader: IsOption]: String => Either[DeserializationOpenAIException, B] = { (s: String) =>
     try
       Right(SnakePickle.read[B](JsonInput.sanitize[B].apply(s)))
     catch {
-      case e: Exception => Left(new DeserializationException(e))
+      case e: Exception => Left(DeserializationOpenAIException(e))
       case t: Throwable =>
         // in ScalaJS, ArrayIndexOutOfBoundsException exceptions are wrapped in org.scalajs.linker.runtime.UndefinedBehaviorError
         t.getCause match {
-          case e: ArrayIndexOutOfBoundsException => Left(new DeserializationException(e))
+          case e: ArrayIndexOutOfBoundsException => Left(DeserializationOpenAIException(e))
           case _                                 => throw t
         }
     }
   }
+
+  def asStringEither: ResponseAs[Either[OpenAIException, String]] =
+    asStringAlways
+      .mapWithMetadata { (string, metadata) =>
+        if (metadata.isSuccess) Right(string) else Left(httpToOpenAIError(HttpError(string, metadata.code)))
+      }
+      .showAs("either(as error, as string)")
+
+  private def httpToOpenAIError(he: HttpError[String]): OpenAIException = {
+    import sttp.model.StatusCode._
+    val errorMessageBody = SnakePickle.read[ujson.Value](he.body).apply("error").obj.values
+    val (message, typ, param, code) =
+      SnakePickle.read[(Option[String], Option[String], Option[String], Option[String])](errorMessageBody)
```

Comment:
**issue:** You are using a tuple here, which means that you assume fields based by their position, right? If that's the case then I think it's unsafe, because the ordering is  not guaranteed and all of them are strings. It's better to deserialize to a case class.